### PR TITLE
Favorite

### PR DESCRIPTION
**Pull request by maksymvasylyev**

### CODE REVIEW

File: components/Favorites/Favorites.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                      src={"/notLiked.png"}
+                      alt="notLiked"
+                      width={40}
+                      height={40}
+                    ></Image>
+                  )}
+                </StyledLikeButton>
+                <Link href={`/detailsCar/${car.id}`}>
+                  <Image
+                    style={{ borderRadius: "10%" }}
```
Comment (mbosselmann) [inline components/Favorites/Favorites.js:37]: We do not want to work with inline stylings. Please use styled components instead.

File: components/Favorites/Favorites.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                    from {car.Price} Euro
+                  </div>
+                </Link>
+                <div
+                  style={{
+                    fontSize: "1.5rem",
+                    color: "black",
+                    position: "relative",
+                    top: "-30px",
+                  }}
```
Comment (mbosselmann) [inline components/Favorites/Favorites.js:66]: same here for inline styling (see comment above)

File: components/Favorites/Favorites.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    alt={car.model}
+                    width={300}
+                    height={150}
+                  />
+                </Link>
+                <Link
+                  href={`/detailsCar/${car.id}`}
+                  style={{ textDecoration: "none" }}
+                >
+                  <div
```
Comment (mbosselmann) [inline components/Favorites/Favorites.js:48]: A p element would be more semantic.

File: components/Favorites/Favorites.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                      color: "white",
+                      position: "relative",
+                      top: "-30px",
+                      left: "20px",
+                    }}
+                  >
+                    from {car.Price} Euro
+                  </div>
+                </Link>
+                <div
```
Comment (mbosselmann) [inline components/Favorites/Favorites.js:60]: A p element would be more semantic too.

File: pages/favorites/index.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    const updatedFavCars = filteredCars.map((car) => {
+      if (car.id === id) {
+        return { ...car, isFavorite: !car.isFavorite };
+      } else return car;
+    });
+    setFilteredCars(updatedFavCars);
+    onToggleFavorite(id);
+  }
+  return (
+    <>
```
Comment (mbosselmann) [inline pages/favorites/index.js:17]: The fragments are not needed.

File: components/FilterForm/FilteredCarsList.js [hunk 6]
```diff
@@ -20,6 +28,23 @@ function FilteredCarsList({ list }) {
...
+                  ></Image>
+                ) : (
+                  <Image
+                    src={"/notLiked.png"}
+                    alt="notLiked"
+                    width={40}
+                    height={40}
+                  ></Image>
+                )}
+              </StyledLikeButton>
```
Comment (mbosselmann) [inline components/FilterForm/FilteredCarsList.js:47]: You could think about creating a new component here as you use this part for your Favorites and your FilteredCarsList. I would highly recommend to do this. ;)

File: components/Favorites/Favorites.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+import StyledList from "../CarsList/StyledList";
+import { StyledLikeButton } from "../FilterForm/FilteredCarsList";
+import Image from "next/image";
+import { StyledResultDiv } from "../FilterForm/FilteredCarsList";
+
+function Favorites({ cars, handleToggleFavorite }) {
+  return (
+    <>
+      {cars.filter((car) => car.isFavorite).length < 1 ? (
+        <StyledResultDiv>{"You didn`t like anything yet"}</StyledResultDiv>
```
Comment (mbosselmann) [inline components/Favorites/Favorites.js:11]: `StyledText` would be a better name I think. 🤔

File: components/FilterForm/FilteredCarsList.js [hunk 8]
```diff
@@ -1,17 +1,25 @@
...
   color: red;
   font-size: 2rem;
   margin: 50px;
 `;
 
-function FilteredCarsList({ list }) {
+export const StyledLikeButton = styled.button`
+  background-color: transparent;
+  border: none;
+  display: flex;
```
Comment (mbosselmann) [inline components/FilterForm/FilteredCarsList.js:16]: Do you need `display: flex`? 🤔

File: components/FilterForm/StyledForm.js [hunk 9]
```diff
@@ -6,6 +6,7 @@ export default styled.form`
   display: flex;
   flex-direction: column;
   align-items: center;
+  margin: 0 auto 0 auto;
```
Comment (mbosselmann) [inline components/FilterForm/StyledForm.js:9]: `margin: 0 auto` should do the same :)

File: components/Favorites/Favorites.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+      ) : (
+        <StyledList>
```
[the remark this answers is not in the document]
Comment (maksymvasylyev) [reply components/Favorites/Favorites.js]: Thank you very much!
Now my code looks much more structured.👍🏼

**Review by djfarly — CHANGES_REQUESTED** [review]
Hej Max,
please see my comment about duplicated state.

Otherwise LGTM :)

**Review by NikRoe — APPROVED** [review]
Hi Max,

this looks very nice! 👍 

I left only two comments:

You don't need to style the Head component, since only meta information will be stored there. I left you a suggestion how you could accomplish this.

And there is a div you called StyledLink. You could rethink this, as right now it is somehow confusing :)

**Pull request merged**